data/statuses.js

```diff
@@ -398,7 +398,7 @@ let BattleStatuses = {
 			}
 			const hitMove = new this.Data.Move(data.moveData);
 
-			this.tryMoveHit(target, data.source, /** @type {ActiveMove} */(hitMove));
+			this.tryMoveHit(target, data.source, /** @type {ActiveMove} */(/** @type {unknown} */(hitMove)));
```

Comment:
This is wrong, or at the very least misleading. Because you've made `ActiveMove` a class you need to correct the callsites to actually have an `ActiveMove` object, not a cast `Move` (or arbitrary object). 

```
$ git grep "@type {ActiveMove}"
data/mods/gen1/moves.js:                                  this.moveHit(target, pokemon, move, /** @type {ActiveMove} */ ({damage: this.effectData.totalDamage * 2}));
data/mods/gen1/scripts.js:                        move = /** @type {ActiveMove} */ ({
data/mods/gen2/moves.js:                                  /** @type {ActiveMove} */
data/mods/gen2/scripts.js:                        move = /** @type {ActiveMove} */ ({
data/mods/gen2/statuses.js:                       move = /** @type {ActiveMove} */ ({
data/mods/gen3/moves.js:                                  /** @type {ActiveMove} */
data/mods/gen3/moves.js:                  let moveData = /** @type {ActiveMove} */ ({
data/mods/gen4/moves.js:                                  let moveData = /** @type {ActiveMove} */ ({
data/mods/gen4/moves.js:                  let moveData = /** @type {ActiveMove} */ ({
data/mods/gen4/moves.js:                  let moveData = /** @type {ActiveMove} */ ({
data/mods/gen6/statuses.js:                       this.damage(damage, pokemon, pokemon, /** @type {ActiveMove} */ ({
data/mods/gennext/moves.js:                                       let moveData = /** @type {ActiveMove} */ ({
data/mods/stadium/scripts.js:                     move = /** @type {ActiveMove} */ ({
data/statuses.js:                 this.damage(damage, pokemon, pokemon, /** @type {ActiveMove} */ ({
data/statuses.js:                 this.tryMoveHit(target, data.source, /** @type {ActiveMove} */(/** @type {unknown} */(hitMove)));
```

Are all places where we are casting which may or may not be OK, but in certain places it definitely isn't: 

```
A battle crashed with this stack trace:
TypeError: move.getHitData is not a function
    at Battle.modifyDamage (/home/ps/showdown/data/mods/gen4/scripts.js:44:23)
    at Battle.getDamage (/home/ps/showdown/.sim-dist/battle.js:2047:15)
    at Battle.onTry (/home/ps/showdown/data/mods/gen4/moves.js:700:22)
    at Battle.singleEvent (/home/ps/showdown/.sim-dist/battle.js:416:25)
    at Battle.tryMoveHit (/home/ps/showdown/data/mods/gen3/scripts.js:182:13)
    at Battle.useMoveInner (/home/ps/showdown/data/mods/gen3/scripts.js:147:18)
    at Battle.useMove (/home/ps/showdown/data/scripts.js:138:25)
    at Battle.runMove (/home/ps/showdown/data/scripts.js:93:31)
    at Battle.runAction (/home/ps/showdown/.sim-dist/battle.js:2557:9)
    at Battle.go (/home/ps/showdown/.sim-dist/battle.js:2777:9)
```